common/script/index.coffee

```diff
@@ -72,13 +72,51 @@ api.daysSince = (yesterday, options = {}) ->
   Math.abs api.startOfDay(_.defaults {now:yesterday}, o).diff(api.startOfDay(_.defaults {now:o.now}, o), 'days')
 
 ###
-  Should the user do this taks on this date, given the task's repeat options and user.preferences.dayStart?
+  Should the user do this task on this date, given the task's repeat options and user.preferences.dayStart?
 ###
-api.shouldDo = (day, repeat, options={}) ->
-  return false unless repeat
+api.shouldDo = (day, dailyTask, options = {}) ->
+  return false unless dailyTask.type == 'daily' && dailyTask.repeat
+  # MIGRATION HACK, FIXME
+  if dailyTask.startDate == null
```

Comment:
I like this, actually. A daily in this new format should always have a start date, right? 

Can we make it `if !dailyTask.startDate` instead to cover all falsy dates?